internal/healthcheck/go_health.go

```diff
@@ -0,0 +1,78 @@
+package healthcheck
+
+import "github.com/InVisionApp/go-health"
+
+type GoHealthService struct {
+	Client HealthClient
+	jobs   []*HealthJob
+}
+
+type HealthClient interface {
+	AddChecks(cfgs []*health.Config) error
+	Start() error
+	Stop() error
+	State() (map[string]health.State, bool, error)
+}
+
+func (s *GoHealthService) Start() error {
+
+	cfgs := []*health.Config{}
+	for _, job := range s.jobs {
+		cfgs = append(cfgs, &health.Config{
+			Name:     job.Name,
+			Checker:  job.Checker,
+			Interval: job.Interval,
+		})
+	}
+	err := s.Client.AddChecks(cfgs)
+	if err != nil {
+		return err
+	}
+
+	return s.Client.Start()
+}
+
+func (s *GoHealthService) Stop() error {
+	return s.Client.Stop()
+}
+
+func (s *GoHealthService) Check() (*CheckResult, error) {
```

Comment:
I find myself do not understand "why" this function exists as well as other methods. Perhaps these are source of my confusion :

- I am still new to health check things
- The codes can improved by documentation such as comment to explain "why" and "goal"

Reply:
Health check: checking current services condition such as their connectivity within other service, connectivity to the internal, disk usage, memory usage, etc